src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/WinformsResultProvider.cs

```diff
@@ -78,8 +82,10 @@ public async IAsyncEnumerable<AnalyzeResult> AnalyzeAsync(AnalyzeContext analysi
                 {
                     yield return new()
                     {
-                        RuleId = this._id,
-                        RuleName = this.Name,
+                        RuleId = RuleId,
+                        RuleName = RuleName,
+                        HelpUri = _helpUri,
+                        FullDescription = FullDescription,
```

Comment:
The ResultMessage comes from IUpdater for eg - https://github.com/dotnet/upgrade-assistant/blob/main/src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/WinformsDefaultFontUpdater.cs - the title, description and message is defined here. I think these needs to be treated similar to diagnostic analyzers and each of the updater implementation should provide the info on RuleId, RuleName, helpURI and FullDescription

Reply:
I think we should change the `IUpdater` in a separate PR.
Otherwise, we will change many files.

What do you think?

Reply:
Yeah, that'll be a bigger change and would be good in a separate PR

Reply:
agreed, we can table the changes to IUpdater as part of the changes to Diagnostic Analyzer. Could we make this PR just for the Dependency Analysis section then @eddynaka ? Could we revert back what we had for the `WinformsResultProvider.cs`?

Reply:
sure.

Reply:
done.